torch/csrc/jit/script/compiler.cpp

```diff
@@ -2003,6 +2003,31 @@ struct to_ir {
   }
 
   void emitAssignment(const Assign& stmt) {
+    static std::atomic<size_t> tmp_count{0};
+    if (stmt.lhs_list().size() == 1) {
+      return emitSingleAssignment(stmt);
+    }
+    // multiple assign & annotated type not supported in python
+    TORCH_INTERNAL_ASSERT(stmt.lhs_list().size() > 1 && !stmt.type().present());
+    // a = b = expr()
+    // the semantics of multiple assignment is that expr() is emitted once, then
+    // from left to right the assignments are made
+    const auto tmp_name =
+        std::string("$tmp_assign_") + std::to_string(tmp_count++);
```

Comment:
We're starting to have a number of places that generate fake names (e.g. for list comprehensions), not necessary for this PR but we should think about consolidating those so we don't end up with collisions

https://github.com/pytorch/pytorch/blob/10c456417ce49cb5bfad8aba2bd7e3a23a83aef2/torch/csrc/jit/script/compiler.cpp#L1043